### DISCUSSION TITLE
Help on Zingg Entity Resolution Workflow on Databricks

### DISCUSSION THREAD

**Discussion by iitbhumanish** [zinggAI/zingg#569] [Q&A] [answered]
Please help me if I miss something . highly appreciated.

I am going through all steps as written in below script.

https://github.com/zinggAI/zingg/blob/main/examples/databricks/FebrlExample.ipynb

Save all the labels provided by the user --> it is done
Build the Zingg models and predict matches --> not sure why it is failing even I have marked matched records

![image](https://user-images.githubusercontent.com/131442577/233655522-bd0c497f-020a-4048-b7d6-34b4784867f2.png)

![image](https://user-images.githubusercontent.com/131442577/233655935-cdc53bec-6c5d-4d2c-973a-b4685255638e.png)

**Comment by sonalgoyal** [comment, marked as answer]
Can you restart your cluster and try? We are seeing some caching issues in the notebook which we need to fix

**Reply by iitbhumanish** [reply]
Thanks a lot for quick response Sonal. 
Could you please tell , after restart  do I need to run all steps ( 1-12) again  . If yes, does it also require to change modelId  to point different location ? Please advise.

 1.Define locations for the model
 2.Setup common functions for use in Zingg.
 3.Start building the Zingg program
 4.Define the input
 5.Configure the output
 6.Define the match fields and their types
 7.Performance settings
 8.Select pairs for user labeling
 9.See if we have records for labeling
 10.Label the pairs
 11.Save all the labels provided by the user
 12.Build the Zingg models and predict matches

**Comment by sonalgoyal** [comment]
No you need to only run the last step

**Reply by iitbhumanish** [reply]
Thanks it worked .  But it generated matched records in csv file and unable to read it using below command.
Should these not be saved as parquet files ? am I missing something.

![image](https://user-images.githubusercontent.com/131442577/233737011-f3883827-006a-467c-8b3e-14e55a45358c.png)

![image](https://user-images.githubusercontent.com/131442577/233738188-779490d8-4a64-4840-bec1-5f5e9261abf5.png)

**Comment by sonalgoyal** [comment]
The output format will depend on what you configure in your program. So please check that

**Reply by iitbhumanish** [reply]
Thanks it worked after modifying the output params
outputPipe = Pipe(name="matched", format="delta")

**Comment by iitbhumanish** [comment]
Any suggestion why I get this error on z_isMatch ? Please advise.

![image](https://user-images.githubusercontent.com/131442577/233831983-f5ae28eb-73de-4906-b54d-5d6d892b0556.png)
